src/xrpld/overlay/detail/OverlayImpl.cpp

```diff
@@ -1224,15 +1248,18 @@ OverlayImpl::relay(
     // We have more peers than the minimum (disabled + minimum enabled),
     // relay to all disabled and some randomly selected enabled that
     // do not have the transaction.
-    auto enabledTarget = app_.config().TX_REDUCE_RELAY_MIN_PEERS +
-        (total - minRelay) * app_.config().TX_RELAY_PERCENTAGE / 100;
+    auto const enabledTarget = relay  //
+        ? app_.config().TX_REDUCE_RELAY_MIN_PEERS +
+            (total - minRelay) * app_.config().TX_RELAY_PERCENTAGE / 100
+        : 0;
 
     txMetrics_.addMetrics(enabledTarget, toSkip.size(), disabled);
 
     if (enabledTarget > enabledInSkip)
         std::shuffle(peers.begin(), peers.end(), default_prng());
 
-    JLOG(journal_.trace()) << "relaying tx, total peers " << peers.size()
+    JLOG(journal_.trace()) << (relay ? "" : "not ")
+                           << "relaying tx, total peers " << peers.size()
```

Comment:
You can also remove the relay check here.
```suggestion
    JLOG(journal_.trace()) << "relaying tx, total peers " << peers.size()
```
(This is also the original code.)